sympy/vector/operators.py

```diff
@@ -0,0 +1,187 @@
+from sympy.core import Expr
+from sympy.vector.coordsysrect import CoordSysCartesian
+from sympy.vector.vector import Vector
+from sympy.vector.deloperator import Del
+
+
+class Gradient(Expr):
+    """
+    Represents unevaluated Gradient.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> s = R.x*R.y*R.z
+    >>> Gradient(s, R)
+    Gradient(R.x*R.y*R.z, R)
+
+    """
+
+    def __new__(cls, scalar):
+        obj = Expr.__new__(cls, scalar)
+        obj._scalar = scalar
+        return obj
+
+    def _get_coord_sys_from_expr(self, scalar):
+        return list(scalar.atoms(CoordSysCartesian))[0]
+
+    def doit(self):
+        return Del(self._get_coord_sys_from_expr(self._scalar)).gradient(self._scalar).doit()
+
+
+class Divergence(Expr):
+    """
+    Represents unevaluated Divergence.
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, Gradient
+    >>> R = CoordSysCartesian('R')
+    >>> v = R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k
+    >>> Divergence(v, R)
+    Divergence(R.y*R.z*R.i + R.x*R.z*R.j + R.x*R.y*R.k, R)
+
+    """
+
+    def __new__(cls, vect):
+        obj = Expr.__new__(cls, vect)
+        obj._vect = vect
+        return obj
+
+    def _get_coord_sys_from_expr(self, vect):
+        return vect._sys
+
+    def doit(self):
+        return Del(self._get_coord_sys_from_expr(self._vect)).dot(self._vect).doit()
```

Comment:
What about putting this code in the functions? I think it's clearer if `doit( )` just calls the respective function (in this case `divergence`).